packages/loader/container-loader/src/container.ts

```diff
@@ -1305,7 +1307,8 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
     }
 
     private async getDocumentStorageService(): Promise<IDocumentStorageService> {
-        return this._deltaManager.connectToStorage();
+        const storage = await this._deltaManager.connectToStorage();
+        return new ProtocolTreeStorageService(storage, () => this.captureProtocolSummary());
```

Comment:
I'd prefer this wrapping to be in DeltaManager.connectToStorage() method, such that it's easier to see all the adapters, their order and when they get inserted. This will also avoid any issues in the future if/when this method is called many times - DeltaManager does caching of resulting object, while this method does not.

Reply:
BTW, I'd remove this method on Container altogether and inline it. There is this.storageService there as well and if you do not  look into them, it might be not clear for reader what's the difference (can easily miss misuse when doing PR)

Reply:
Any thoughts on lifting the implementation of `.connectToStorage()` out of `DeltaManager` entirely and up to here?  I think I would prefer it since `DeltaManager` doesn't use the storage directly -- it seems a bit strange for `DeltaManager` to be the middle-man and hold the policy for acquiring the `IDocumentStorageService`.

Reply:
i investigated this, and it's not feasible without breaking the existing separation of concerns, as it would required getting the protocol handler into the delta manager. The existing abstraction is incorrect, as i don't think the containers storage should be controlled by delta manager at all, it's unclear why delta manger is involved in it's creation flow at all because it never even uses the storage